lib/generators/rspec/integration/templates/api_request_spec.rb

```diff
@@ -11,59 +11,65 @@
 # It only uses APIs available in rails and/or rspec-rails.  There are a number
 # of tools you can use to make these specs even more expressive, but we're
 # sticking to rails and rspec-rails APIs to keep things simple and stable.
-#
-# Compared to earlier versions of this generator, there is very limited use of
-# stubs and message expectations in this spec.  Stubs are only used when there
-# is no simpler way to get a handle on the object needed for the example.
-# Message expectations are only used when there is no simpler way to specify
-# that an instance is receiving a specific message.
-#
-# Also compared to earlier versions of this generator, there are no longer any
-# expectations of assigns and templates rendered. These features have been
-# removed from Rails core in Rails 5, but can be added back in via the
-# `rails-controller-testing` gem.
+
 
 <% module_namespacing do -%>
-RSpec.describe <%= controller_class_name %>Controller, <%= type_metatag(:controller) %> do
+RSpec.describe "<%= class_name.pluralize %>", <%= type_metatag(:request) %> do
 
   # This should return the minimal set of attributes required to create a valid
   # <%= class_name %>. As you add validations to <%= class_name %>, be sure to
   # adjust the attributes here as well.
-  let(:valid_attributes) {
+  let(:valid_attributes) do
+<% if options[:factorybot] || Gem.loaded_specs.has_key?('factory_bot_rails') -%>
+     attributes_for(:<%= ns_file_name -%>)
+<% elsif options[:fabrication] -%>
+    Fabricate.attributes_for(<%= ns_prefix.empty? ? ":"+ns_file_name : "'#{class_name}'"-%>)
+<% else -%>
     skip("Add a hash of attributes valid for your model")
-  }
+<% end -%>
+  end
```

Comment:
You can remove all of the above as per the earlier comment, we don't add integration for 3rd parties here, thats for them to do.